src/app/account/mfa/sms/page.tsx

```diff
@@ -1,199 +1,186 @@
-    }
-
-    return (
-        <>
-            <h2>Phone numbers reserved for MFA</h2>
-            <ul>
-                {mfaPhones.map((phone) => {
-                    return (
-                        <li key={phone.id} style={{ display: 'flex', gap: '10px' }}>
-                            <p>
-                                {phone.phoneNumber} {phone.defaultSecondFactor && '(Default)'}
-                            </p>
-                            <div>
-                                <Button onClick={() => phone.setReservedForSecondFactor({ reserved: false })}>
-                                    Disable for MFA
-                                </Button>
-                            </div>
-
-                            {!phone.defaultSecondFactor && (
-                                <div>
-                                    <Button onClick={() => phone.makeDefaultSecondFactor()}>Make default</Button>
-                                </div>
-                            )}
-
-                            {user.phoneNumbers.length > 1 && (
-                                <div>
-                                    <Button onClick={() => phone.destroy()}>Remove from account</Button>
-                                </div>
-                            )}
-                        </li>
-                    )
-                })}
-            </ul>
-            You have enabled MFA on your account
-            <Link href="/">
-                <Button>Return to homepage</Button>
-            </Link>
-        </>
-    )
-}
-
-// Display phone numbers that are not reserved for MFA
-const ManageAvailablePhoneNumbers = () => {
-    const { user } = useUser()
-
-    if (!user) return null
-
-    // Reserve a phone number for MFA
-    const reservePhoneForMfa = async (phone: PhoneNumberResource) => {
-        // Set the phone number as reserved for MFA
-        await phone.setReservedForSecondFactor({ reserved: true })
-        // Refresh the user information to reflect changes
-        await user.reload()
-    }
-
-    // phone numbers are valid for MFA but aren't used for it
-    const availableForMfaPhones = user.phoneNumbers.filter(
-        (phone) => phone.verification.status === 'verified' && !phone.reservedForSecondFactor,
-    )
+export default function ManageSMSMFA() {
+    const { isLoaded, user } = useUser()
+    const [phoneNumber, setPhoneNumber] = React.useState('')
+    const [verificationCode, setVerificationCode] = React.useState('')
+    const [error, setError] = React.useState('')
+    const [verificationSuccess, setVerificationSuccess] = React.useState(false)
+    const [codeSent, setCodeSent] = React.useState(false)
+    const [isSendingCode, setIsSendingCode] = React.useState(false)
+    const [isVerifying, setIsVerifying] = React.useState(false)
+    const [phoneResourceToVerify, setPhoneResourceToVerify] = React.useState<PhoneNumberResource | null>(null)
```

Comment:
There is a lot of state in this single component which seems to get pretty complicated in the JSX, i wonder if we can extract some of these out into their own smaller components so its more easily testable? one immediate area i notice is the three different conditions in the jsx for `!verificationSuccess &&` and `verificationSuccess &&`. though maybe it doesn't make sense to over abstract this stuff since we don't really need to reuse it anywhere else..